MapboxNavigation/MapboxSpeechController.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+import MapboxSpeech
+import AVFoundation
+import MapboxCoreNavigation
+import CoreLocation
+import MapboxDirections
+
+/**
+ `PollyVoiceController` extends the default `RouteVoiceController` by providing support for AWSPolly. `RouteVoiceController` will be used as a fallback during poor network conditions.
+ */
+@objc(MBMapboxSpeechController)
+public class MapboxSpeechController: RouteVoiceController {
+    
+    /**
+     Number of seconds a Polly request can wait before it is canceled and the default speech synthesizer speaks the instruction.
+     */
+    @objc public var timeoutIntervalForRequest:TimeInterval = 5
+    
+    /**
+     Number of steps ahead of the current step to cache spoken instructions.
+     */
+    @objc public var stepsAheadToCache: Int = 3
+    
+    /**
+     Locale used to decide on language for spoken instruction.
+     
+     By default, the users device will be used to decide on the locale.
+     */
+    @objc public var locale: Locale?
+    
+    let sessionConfiguration = URLSessionConfiguration.default
+    var urlSession: URLSession
+    var cacheURLSession: URLSession
+    var pollyTask: URLSessionDataTask?
+    var cachePollyTask: URLSessionDataTask?
+    var spokenInstructionsForRoute = NSCache<NSString, NSData>()
+    
+    var mapboxSpeechSynth: SpeechSynthesizer
```

Comment:
Avoid abbreviating “synthesizer” as “synth”. It’s also a bit redundant that the class <strong>Mapbox</strong>SpeechController in the module <strong>Mapbox</strong>Navigation contains the property <strong>mapbox</strong>SpeechSynth.